core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -302,24 +312,19 @@ public String getRunnerSuffix() {
     }
 
     @ConfigGroup
-    public static class VineFlowerConfig {
+    public static class DecompilerConfig {
         /**
          * An advanced option that will decompile generated and transformed bytecode into the 'decompiled' directory.
          * This is only taken into account when fast-jar is used.
          */
-        @ConfigItem(defaultValue = "false")
-        public boolean enabled;
-
-        /**
-         * The version of Vineflower to use
-         */
-        @ConfigItem(defaultValue = "1.9.3")
-        public String version;
+        @ConfigDocDefault("false")
+        public Optional<Boolean> enabled;
 
         /**
          * The directory into which to save the Vineflower tool if it doesn't exist
          */
-        @ConfigItem(defaultValue = "${user.home}/.quarkus")
-        public String jarDirectory;
+        @ConfigItem
+        @ConfigDocDefault("${user.home}/.quarkus")
+        public Optional<String> jarDirectory;
```

Comment:
It seems like this whole configuration should be separate from packaging. For example if we produce multiple artifacts with different packaging configurations, we'd still only produce one set of decompiled sources, right?

Reply:
Yeah, I'm not against that.